src/Compilers/CSharp/Test/Semantic/Diagnostics/DiagnosticAnalyzerTests.cs

```diff
@@ -945,12 +945,12 @@ public override void Initialize(AnalysisContext context)
             }
         }
 
-        [Fact, WorkItem(252, "https://github.com/dotnet/roslyn/issues/252")]
+        [Fact, WorkItem(252, "https://github.com/dotnet/roslyn/issues/252"), WorkItem(1392, "https://github.com/dotnet/roslyn/issues/1392")]
         public void TestReportingUnsupportedDiagnostic()
         {
             string source = @"";
             var analyzers = new DiagnosticAnalyzer[] { new AnalyzerReportingUnsupportedDiagnostic() };
-            string message = new ArgumentException("Reported diagnostic with ID 'ID_2' is not supported by the analyzer.", "diagnostic").Message;
+            string message = new ArgumentException(string.Format(AnalyzerDriverResources.UnsupportedDiagnosticReported, AnalyzerReportingUnsupportedDiagnostic.UnsupportedDescriptor.Id), "diagnostic").Message;
```

Comment:
This is the origin of the exception that is converted into a diagnostic: http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalysisContextHelpers.cs,37